src/Utilities/Compiler/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -193,6 +193,16 @@ private static bool HasDisposeMethodSignature(this IMethodSymbol method)
                 method.ReturnsVoid && method.Parameters.IsEmpty;
         }
 
+        /// <summary>
+        /// Checks if the given method matches Dispose method convention and can be recognized by "using".
+        /// </summary>
+        public static bool HasDisposeSignatureByConvention(this IMethodSymbol method)
+        {
+            return method.HasDisposeMethodSignature()
+                && !method.GetSymbolModifiers().HasFlag(SymbolModifiers.Static)
+                && !method.IsPrivate();
```

Comment:
Is the `Dispose` convention working for something else than `public`?

Reply:
Apparently, the answer is yes. This compiles:
```csharp
ref struct InternallyDisposable { internal void Dispose() { } }
...
using var internallyDisposable = new InternallyDisposable();
```

Reply:
Might be worth adding a test for this case too to avoid regression

Reply:
Added one for `internal`. `protected` is undefined/impossible at the moment too, as structs can't declare new `protected` members.